code/src/context/I18nProvider.tsx

```diff
@@ -1,18 +1,67 @@
-import { FC, ReactNode } from "react";
+import React, {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  createContext,
+  useContext,
+  useLayoutEffect,
+  useState
+} from "react";
 import { IntlProvider } from "react-intl";
 
+import { LOCAL_STORAGE_KEYS } from "@/constants/common";
 import messages from "@/messages";
 
-type I18nProividerProps = {
-  children: ReactNode;
+type I18nProviderProps = Required<PropsWithChildren>;
+
+export type Locale = "en" | "uk";
+
+type LocaleContextType = {
+  locale: Locale;
+  setLocale: Dispatch<SetStateAction<Locale>>;
 };
 
-const I18nProivider: FC<I18nProividerProps> = ({ children }) => {
+const LocaleContext = createContext<LocaleContextType | null>(null);
+
+const I18nProvider = ({ children }: I18nProviderProps) => {
+  const [locale, setLocale] = useState<Locale>("en");
+
+  useLayoutEffect(() => {
```

Comment:
Is `useLayoutEffect` required here? Maybe just pass function that returns initial language as initial value to `useState`?

Reply:
Thank you for the idea. Fixed!